### ISSUE TITLE
Possible error in code

### ISSUE THREAD

**Issue by mlucas491** [eriklindernoren/PyTorch-GAN#25]
In line 265 the code does not look right :

code_input = Variable(FloatTensor(np.random.normal(-1, 1, (batch_size, opt.code_dim))))

instead of 'normal' shouldn't be 'uniform'?

- Mirtha

**Comment by mlerma54** [comment]
Just to clarify, Mirtha's question is about the implementation of **infoGAN**.

Also I noticed that at no point the (infoGAN) network is given information that allows it to associate labels to digits, it looks like "labels" is being treated as an additional code variable (but categorical rather than numeric). So the network manages to cluster the digits into 10 classes, but with no control of what digit goes to which class, they end up in a random order - and what is worse, the same class may contain hybrid versions of digits that look similar such as 4 and 9. This can be fixed with small changes in the loss functions for generator and discriminator - if I have a chance I may post my suggested changes at a later time.

**Comment by eriklindernoren** [comment]
@mlucas491 Thanks, that should definitely be `np.random.uniform`.

@mlerma54 You're observation is correct, and that is also one of the interesting properties with InfoGAN. The network finds it's own clustering based on the properties of the data, in a totally unsupervised manner.

**Comment by eriklindernoren** [comment]
See [fix](https://github.com/eriklindernoren/PyTorch-GAN/commit/98e1b5addc77621b9137b743c85b8d1baeb2e58a).
